samza-kv/src/main/java/org/apache/samza/storage/kv/BaseLocalStoreBackedTableDescriptor.java

```diff
@@ -57,9 +70,57 @@ public D withSideInputsProcessor(SideInputsProcessor sideInputsProcessor) {
     return (D) this;
   }
 
+  /**
+   * Disable changelog for this table, by default changelog is enabled.
+   * Refer to <code>stores.store-name.changelog</code> in Samza configuration guide
+   *
+   * @return this table descriptor instance
+   */
+  public D withChangelogDisabled() {
+    this.enableChangelog = false;
+    return (D) this;
+  }
+
+  /**
+   * Specify the changelog stream name, by default changelog stream name is automatically
+   * generated in format [job-name]-[job-id]-table-[table-id]. For both auto-generated
+   * and user specified changelog stream, invalid characters will be automatically
+   * replaced with '-' to confirm with pattern { @literal [a-zA-Z0-9_-]+ }
+   *
+   * Refer to <code>stores.store-name.changelog</code> in Samza configuration guide
+   *
+   * @param changelogStream changelog stream name
+   * @return this table descriptor instance
+   */
+  public D withChangelogStream(String changelogStream) {
```

Comment:
Do we have a use-case for overriding the auto-generated `changelogStream`? 
Is it possible to throw an exception instead of rewriting the changelog stream if it doesn't conform with the pattern `[a-zA-Z0-9_-]+`?

Reply:
Since this is more of an internal feature, wouldn't it more user friendly to auto-repair it?

Reply:
@weisong44 I'd prefer to throw an exception here as well. If the auto-generated name is invalid, it would imply that the job name is invalid. Job name is used in other places to create streams (e.g., coordinator, checkpoint) so that'd still be an issue, and users should fix that. 

In general, I'd prefer to not automatically rewrite names - it causes confusion since the name you'd expect from code or convention doesn't match the actual name. For example, this is an issue with some metrics reporters that "normalize" dashes in the metric name to underscores - it's very confusing to figure out the actual metric name.

Reply:
Based on what we discussed